Islandora/islandora_batch
# 7.x islandora 2046: Allow Islandora Batch to be extended for more fun

**JIRA Ticket**: (https://jira.duraspace.org/browse/ISLANDORA-2046)

# What does this Pull Request do?

Allows Islandora Batch to be used for other more repository actions (like delete or update) but keeps  "ingest" as default to allow any other extending class of IslandoraBatchObject to keep working as before. Also changes `islandora_batch_queue`primary index to a compound one, to allow same Islandora installation to keep same PID in different sets, but, still enforces uniqueness inside a single SET.

# What's new?
If you code a little, your Islandora Batch could be able to update, edit, drop, purge, clone or QA your objects. Up to you. If you don't code, all is still the same in Islandora paradise ! 🌴 
But: this opens a wide open world to reusing our batch queue for post-processing, editing, etc operations, including working on existing objects using pieces from newly created in-batch ones.
Also: this is  a first step, probably more can be done and will be done in the future.

# How should this be tested?

Please make sure that your existing Batch Sets and New ingests still work. Not much more?

Enjoy/fear the flexibility.

[Islandora IMI ](https://github.com/mnylc/islandora_multi_importer/blob/e674a8371efddeba7b4ca839697a69a5e3a1b29b/includes/islandora_multi_batch.inc#L624-L653 )uses this patch already for its update capabilities. As you see implementation is quite simple. 

# Additional Notes:
* Does this change require documentation to be updated? No
* Does this change add any new dependencies?  No
* Does this change require any other modifications to be made to the repository (ie. Regeneration activity, etc.)?  No
* Could this change impact execution of existing code? Nope. 

# Interested Parties
@Islandora/7-x-1-x-committers @adam-vessey @jordandukart @jonathangreen
